### ISSUE TITLE
Public HD key export shares private key

### ISSUE THREAD

**Issue by craigraw** [BlockchainCommons/GordianSeedTool-iOS#182]
As the title suggests, the `ur:crypto-hdkey` shared from Derive Key > Other Key Derivations > Public HD Key actually shares the private key information.

**Comment by ChristopherA** [comment]
Assigned to @wolfmcnally to evaluate and @ChristopherA to prioritize.

**Comment by craigraw** [comment]
Apologies, I was incorrect - `ur:crypto-hdkey` does not share the private key.

However, I think there is still a problem. The UR is encoded with the `master` field as `true`, however [the specification states](https://github.com/BlockchainCommons/Research/blob/master/papers/bcr-2020-007-hdkey.md#cddl-for-hdkey) that:

```
; A master key is always private, has no use or derivation information,
; and always includes a chain code.
```

Therefore hummingbird was creating an object that represented a private key - I believe correctly?

**Comment by wolfmcnally** [comment]
I'm a bit confused: are you saying there is a problem with how SeedTool is encoding non-master URs?

**Comment by craigraw** [comment]
Yes - specifically that a public key is encoding with the master field set to true, in contravention of the spec.

**Comment by wolfmcnally** [comment]
Thanks for calling this to my attention. I'm flagging this issue as a bug.

**Comment by craigraw** [comment]
Should this be fixed? I tested using GST 1.6 (66) and found the public key is still being shared with the master field set to true.

**Comment by shannona** [comment]
This seems irrelevent as this is now shared as a `ur:envelope` rather than a `ur:crypto-hdkey`.

This is the Public HD Key from Yinmn Blue, derived from a Master Key:
```
ur:envelope/lntpcshdclaomucnprrerhldtidtkokkoszogolatlpkdintpfdkgygtbwzckigtmerpcpatstgsoycfaddwlfcfaddpoycfadmhcfadmeoycfadyktpcshdcxvdbggasekkuytykkfefdlahtflkecpbzyackrpfmghvyqdssfsfhmtzcttsehdcsoyadcssgoyadcfadwkoybdtpcskscxfdfygrihkkcxiyjpjljncxfyhsjpjecxgdkpjpjojzihcxgdihiajecxhfinhsjzlrolwsts
```
Envelope describes it as such:
```
Bytes(33) [
    isA: BIP32Key
    isA: PublicKey
    asset: BTC [
        network: MainNet
    ]
    chainCode: Bytes(32)
    hasName: "HDKey from Dark Purple Peck Vial"
]
```
Transforming the Envelope's bytewords to hex results in this:
```
86d8185821029323b2b5b989d0297679a7fb5580d5aa279db024514d13fd7d4d91b62207c74ca119012c8219012da1190190190191a11901f5d8185820e71249c179dbd4794548805a477c2215f81eb63e54e1b3c43d3f96fdd1c15818a10118caa1011901f4a10bd818782048444b65792066726f6d204461726b20507572706c65205065636b205669616c
```
Which `cbor2diag` reads as follows:
```
[24(h'029323b2b5b989d0297679a7fb5580d5aa279db024514d13fd7d4d91b62207c74c'), {300: [301, {400: 401}]}, {501: 24(h'e71249c179dbd4794548805a477c2215f81eb63e54e1b3c43d3f96fdd1c15818')}, {1: 202}, {1: 500}, {11: 24("HDKey from Dark Purple Peck Vial")}]
```
Any other thoughts, @craigraw ? (We're now at 74.)

**Comment by craigraw** [comment]
I guess if not relevant anymore, it can be closed. Fwiw I have missed the ability to test certain URs in Seed Tool.

**Comment by ChristopherA** [comment]
> I have missed the ability to test certain URs in Seed Tool.

Which ones? We can always enable some when "Developer Mode" is turned on in settings. Let us know.
